CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+##################################################################
+# CMake file for Modelica Buildings Library
+# To build all targets, run
+# rm -rf build && mkdir build && cd build && cmake ../ && cmake --build . --target install && cd ..
+#
+##################################################################
+cmake_policy(SET CMP0048 NEW)
+
+project( Modelica-Buildings VERSION 5.1.0 )
+cmake_minimum_required(VERSION 3.8)
+
+# find_package(Java REQUIRED)
+# include(UseJava)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -ldl")
+
+set( BUILDINGS_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/Buildings")
+
+set( CMAKE_INSTALL_PREFIX "${BUILDINGS_INSTALL_PREFIX}/Resources/Library/${PLATFORM_INSTALL_PREFIX}/" )
+
+# Output build system directories.
+message("CMake source directory is: ${CMAKE_SOURCE_DIR}")
+
+# Detect OS.
+if(CMAKE_SYSTEM_NAME MATCHES Windows)
+    set(WINDOWS 1)
+    message("Windows system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Linux)
+    set(LINUX 1)
+    message("Linux system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Darwin)
+    set(MACOSX 1)
+    message("Mac OS X system detected.")
+else()
+    message(WARNING "OS unknown!")
+endif()
+
+# Set target directories.
+# NOTE: always add quotes to protect spaces in path when setting new variables.
+if (WINDOWS)
```

Comment:
Consider...

```
if(WIN32)
...
elseif(APPLE)
...
elseif(UNIX)
...
else()
  message(WARNING...)
endif()
```

These are all defined natively by cmake and you can avoid the system name tests above.
https://cmake.org/cmake/help/v3.0/variable/WIN32.html
https://cmake.org/cmake/help/v3.0/variable/UNIX.html
https://cmake.org/cmake/help/v3.0/variable/APPLE.html

A common idiom to test for linux is `if(UNIX AND NOT APPLE)` but here the logic makes that unnecessary.